File: VB/MyXtraGrid/Form1.vb
Imports System
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Controls

Namespace MyXtraGrid

    Partial Public Class Form1
        Inherits Form

        Private Function CreateTable(ByVal RowCount As Integer) As DataTable
            Dim tbl As DataTable = New DataTable()
            tbl.Columns.Add("Name", GetType(String))
            tbl.Columns.Add("ID", GetType(Integer))
            tbl.Columns.Add("Number", GetType(Integer))
            tbl.Columns.Add("Date", GetType(Date))
            For i As Integer = 0 To RowCount - 1
                tbl.Rows.Add(New Object() {String.Format("Name{0}", i), i, 3 - i, Date.Now.AddDays(i)})
            Next

            Return tbl
        End Function

        Public Sub New()
            InitializeComponent()
            myGridControl1.DataSource = CreateTable(20)

            Dim showCustomButton As CheckButton = myGridView1.FindPanelItems.AddCheckButton("test", True)
            showCustomButton.ImageOptions.ImageUri.Uri = "Filter;Size16x16;Svg"
            AddHandler showCustomButton.CheckedChanged, AddressOf ShowCustomButton_CheckedChanged
        End Sub

        Private Sub ShowCustomButton_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs)
            MessageBox.Show("CheckButton is clicked")
        End Sub

    End Class
End Namespace

File: CS/MyXtraGrid/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Controls;

namespace MyXtraGrid {
    public partial class Form1 : Form {

        private DataTable CreateTable(int RowCount)
        {
            DataTable tbl = new DataTable();
            tbl.Columns.Add("Name", typeof(string));
            tbl.Columns.Add("ID", typeof(int));
            tbl.Columns.Add("Number", typeof(int));
            tbl.Columns.Add("Date", typeof(DateTime));
            for (int i = 0; i < RowCount; i++)
                tbl.Rows.Add(new object[] { String.Format("Name{0}", i), i, 3 - i, DateTime.Now.AddDays(i) });
            return tbl;
        }
        

        public Form1() {
            InitializeComponent();
            myGridControl1.DataSource = CreateTable(20);

            CheckButton showCustomButton = myGridView1.FindPanelItems.AddCheckButton("test", true);
            showCustomButton.ImageOptions.ImageUri.Uri = "Filter;Size16x16;Svg";
            showCustomButton.CheckedChanged += ShowCustomButton_CheckedChanged;
        }

        private void ShowCustomButton_CheckedChanged(object sender, EventArgs e)
        {
              MessageBox.Show("CheckButton is clicked");
        }

    }
}
